inotify.go

```diff
@@ -101,9 +125,14 @@ func (w *Watcher) Add(name string) error {
 
 // Remove stops watching the the named file or directory (non-recursively).
 func (w *Watcher) Remove(name string) error {
-	name = filepath.Clean(name)
 	w.mu.Lock()
 	defer w.mu.Unlock()
+	return w.remove(name)
+}
+
+// Only call this when w.mu is locked, like from Remove or Close.
```

Comment:
I'm not convinced fine-grained locking is an option here. It seems a trap for race conditions at first glance. Maybe it's even a mistake in Add.